frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -523,10 +524,14 @@ function DashboardInner(props: DashboardProps) {
                   <FilterApplyButton />
                 </ParametersWidgetContainer>
               )}
-
-              <CardsContainer id="Dashboard-Cards-Container">
-                {renderContent()}
-              </CardsContainer>
+              <FixedWidthContainer
+                data-testid="fixed-width-container"
+                isFixedWidth={dashboard?.width === "fixed"}
+              >
+                <CardsContainer id="Dashboard-Cards-Container">
+                  {renderContent()}
+                </CardsContainer>
+              </FixedWidthContainer>
```

Comment:
Wrap `FixedWidthContainer` around the dashboard cards body. Use `dashboard.width` attribute to dynamically set a fixed/responsive width.